sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarStorageStrategy.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import io.opentelemetry.sdk.metrics.view.Aggregation;
+
+/** A strategy for storing exemplars. */
+@FunctionalInterface
+public interface ExemplarStorageStrategy {
+  /** Constructs a new sampling reservoir for a given aggregator. */
+  ExemplarReservoir createReservoir(Aggregation aggregator);
+
+  /** A sampler which will never store Exemplars. */
+  static final ExemplarStorageStrategy ALWAYS_OFF = (agg) -> ExemplarReservoir.empty();
```

Comment:
Probably in the same class as the above comment, let's move these into a package private class and expose them through static methods for consistency with other interfaces